google/cloud/bigtable/internal/table.h

```diff
@@ -195,6 +176,28 @@ class Table {
   //@}
 
  private:
+  //@{
+  /// @name Helper functions to implement constructors with changed policies.
+  void ChangePolicy(RPCRetryPolicy& policy) {
+    rpc_retry_policy_ = policy.clone();
+  }
+
+  void ChangePolicy(RPCBackoffPolicy& policy) {
+    rpc_backoff_policy_ = policy.clone();
+  }
+
+  void ChangePolicy(IdempotentMutationPolicy& policy) {
+    idempotent_mutation_policy_ = policy.clone();
+  }
+
+  template <typename Policy, typename... Policies>
+  void ChangePolicies(Policy&& policy, Policies&&... policies) {
```

Comment:
Nice, I like the functional-style list processing using C++ templates here, I guess this is where `&&` is really useful.

Reply:
Indeed, perfect forwarding FTW.